app/cdap/components/hydrator/components/TopPanel/NameAndDescription/index.tsx

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  CancelButton,
+  CustomTooltip,
+  DescriptionTextField,
+  HydratorMetadata,
+  MetadataLeft,
+  NameTextField,
+  PipelineDescription,
+  PipelineName,
+  PipelineType,
+  SaveButton,
+  StyledIconSVG,
+} from './styles';
+
+interface IGlobalObj {
+  etlRealtime?: string;
+  etlDataStreams?: string;
+  etlBatchPipelines?: string[];
+}
+
+interface IMetadata {
+  description: string;
+  name: string;
+}
+
+interface IArtifact {
```

Comment:
Note for future: We need to start centralizing type definitions for back-end-defined types. `IArtifact` or similar is defined in multiple places.

Reply:
I do notice that. And in different places `Artifact` means a bit different too...